src/script/builder/tir/prim_func_frame.cc

```diff
@@ -62,16 +68,88 @@ tvm::tir::Var Arg(String name, tvm::tir::Var var) {
   return var;
 }
 
-tvm::tir::Buffer Arg(String name, tvm::tir::Buffer buffer) {
+tvm::tir::Var Arg(String name, tvm::tir::Buffer buffer) {
   using namespace tvm::tir;
   Namer::Name(buffer, name);
   PrimFuncFrame frame = Builder::Current()->FindFrame<PrimFuncFrame>().value();
   Var handle(buffer->name + "_handle", DataType::Handle());
   frame->args.push_back(handle);
   frame->buffer_map.Set(handle, buffer);
-  return buffer;
+  return handle;
+}
+
+DictAttrs FuncAttrs(DictAttrs attrs) {
+  using namespace tvm::tir;
+  PrimFuncFrame frame = Builder::Current()->FindFrame<PrimFuncFrame>().value();
+  frame->attrs = attrs;
+  return attrs;
 }
 
+tvm::Type FuncRet(tvm::Type ret_type) {
+  PrimFuncFrame frame = Builder::Current()->FindFrame<PrimFuncFrame>().value();
+  frame->ret_type = ret_type;
+  return ret_type;
+}
+
+tvm::tir::Buffer MatchBuffer(ObjectRef param, Array<PrimExpr> shape, DataType dtype,
+                             tvm::tir::Var data, Array<PrimExpr> strides, PrimExpr elem_offset,
+                             String storage_scope, int align, int offset_factor,
+                             String buffer_type_str, Array<IntImm> axis_separators, Span span) {
+  using namespace tvm::tir;
+  if (data.same_as(NullValue<Var>())) {
+    DataType storage_dtype = dtype;
```

Comment:
i didn't check the implementation very carefully. going to have full trust here, and we can always do quite fixes when bugs pop up